src/pkgcheck/checks/acct.py

```diff
@@ -0,0 +1,127 @@
+import re
+
+from collections import defaultdict
+from itertools import groupby
+from operator import attrgetter, itemgetter
+
+from .. import base
+
+
+class MissingAccountIdentifier(base.Warning):
+    """UID/GID can not be found in account package."""
+
+    __slots__ = ("category", "package", "version", "var")
+
+    threshold = base.versioned_feed
+
+    def __init__(self, pkg, var):
+        super().__init__()
+        self._store_cpv(pkg)
+        self.var = var
+
+    @property
+    def short_desc(self):
+        return (
+            f"unable to determine the value of {self.var} variable")
+
+
+class ConflictingAccountIdentifier(base.Error):
+    """Same UID/GID is used by two different users/groups."""
+
+    __slots__ = ("category", "package", "version", "kind", "identifier", "others")
+
+    threshold = base.versioned_feed
+
+    _sorter = staticmethod(itemgetter(0))
+
+    def __init__(self, pkg, kind, identifier, others):
+        super().__init__()
+        self._store_cpv(pkg)
+        self.kind = kind
+        self.identifier = identifier
+        self.others = tuple(sorted(p.cpvstr for p in others))
+
+    @property
+    def short_desc(self):
+        return (
+            f"{self.kind} id {self.identifier} is already used by other "
+            f"{self.kind}: ({', '.join(self.others)})")
+
+
+class OutOfRangeAccountIdentifier(base.Error):
+    """UID/GID outside allowed allocation range."""
+
+    __slots__ = ("category", "package", "version", "kind", "identifier")
+
+    threshold = base.versioned_feed
+
+    def __init__(self, pkg, kind, identifier):
+        super().__init__()
+        self._store_cpv(pkg)
+        self.kind = kind
+        self.identifier = identifier
+
+    @property
+    def short_desc(self):
+        return (
+            f"{self.kind} id {self.identifier} is outside the permitted "
+            f"static allocation range (0..499, 60001+)")
+
+
+class AcctCheck(base.Template):
+    """Check for acct-* packages.
+
+    Verify that acct-* packages do not use conflicting, invalid or out-of-range
+    UIDs/GIDs.
+    """
+
+    scope = base.repository_scope
+    feed_type = base.package_feed
+    known_results = (MissingAccountIdentifier, ConflictingAccountIdentifier,
+                     OutOfRangeAccountIdentifier)
+
+    repo_grabber = attrgetter("repo")
+
+    def __init__(self, options):
+        super().__init__(options)
+        self.id_re = re.compile(
+            r'ACCT_(?P<var>USER|GROUP)_ID=(?P<quot>[\'"]?)(?P<id>[0-9]+)(?P=quot)')
+        self.seen_uids = defaultdict(lambda: defaultdict(list))
+        self.seen_gids = defaultdict(lambda: defaultdict(list))
+
+    def feed(self, full_pkgset):
+        # TODO: can we filter this earlier?
```

Comment:
Note this TODO. The check would be much faster if it didn't have to skip all the other packages explicitly.

Reply:
At present, it would require writing a custom source class that would feed only what the check requested. That shouldn't be too hard since I recently extended pkgcheck to add a filtered source (respects package.mask) and a git commits source (only yields pkgs that have been changed locally in git). It would also make me have to figure out how to merge source results back into a sane order (which I was saving for parallelization work when the issue is much more prevalent.)